node/hack/repair/repair.go

```diff
@@ -0,0 +1,306 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"flag"
+	"fmt"
+	"github.com/certusone/wormhole/node/pkg/db"
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+	nodev1 "github.com/certusone/wormhole/node/pkg/proto/node/v1"
+	"github.com/certusone/wormhole/node/pkg/vaa"
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+	"golang.org/x/time/rate"
+	"google.golang.org/grpc"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	solanaEmitters = []string{
+		"0def15a24423e1edd1a5ab16f557b9060303ddbab8c803d2ee48f4b78a1cfd6b",
+		"b2dd468c9b8c80b3dd9211e9e3fd6ee4d652eb5997b7c9020feae971c278ab07",
+		"ec7372995d5cc8732397fb0ad35c0121e0eaa90d26f828a534cab54391b3a4f5",
+	}
+
+	solanaRPC  = flag.String("solanaRPC", "http://localhost:8899", "Solana RPC address")
+	stateDir   = flag.String("stateDir", "./repairState", "State directory")
+	adminRPC   = flag.String("adminRPC", "/run/guardiand/admin.socket", "Admin RPC address")
+	solanaAddr = flag.String("solanaProgram", "worm2ZoG2kUd4vFXhvjh93UUH596ayRfgQ2MgjNMTth", "Solana program address")
+)
+
+const (
+	postMessageInstructionID = 0x01
+)
+
+func getAdminClient(ctx context.Context, addr string) (*grpc.ClientConn, error, nodev1.NodePrivilegedServiceClient) {
+	conn, err := grpc.DialContext(ctx, fmt.Sprintf("unix:///%s", addr), grpc.WithInsecure())
+
+	if err != nil {
+		log.Fatalf("failed to connect to %s: %v", addr, err)
+	}
+
+	c := nodev1.NewNodePrivilegedServiceClient(conn)
+	return conn, err, c
+}
+
+var publicRPCEndpoints = []string{
+	"https://wormhole-v2-mainnet-api.certus.one",
+	"https://wormhole.inotel.ro",
+	"https://wormhole-v2-mainnet-api.mcf.rocks",
+	"https://wormhole-v2-mainnet-api.chainlayer.network",
+	"https://wormhole-v2-mainnet-api.staking.fund",
+	"https://wormhole-v2-mainnet.01node.com",
+}
+
+func main() {
+	flag.Parse()
+
+	ctx := context.Background()
+	sr := rpc.New(*solanaRPC)
+
+	conn, err, admin := getAdminClient(ctx, *adminRPC)
+	defer conn.Close()
```

Comment:
Don't you normally do defer after the err check?

Reply:
Indeed